gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -309,13 +360,26 @@ def _process_header_lines(self, header_lines):
       # We need to keep the last line of the header from the file because it
       # contains the sample IDs, which is unique per file.
       header_lines = self._representative_header_lines + header_lines[-1:]
-    self._init_with_header(header_lines)
+    if isinstance(self, PySamParser):
+      if header_lines and not header_lines[0].startswith(
+          FILE_FORMAT_HEADER_TEMPLATE.format(VERSION='')):
+        header_lines.insert(
+            0, FILE_FORMAT_HEADER_TEMPLATE.format(VERSION='4.0'))
+      header_lines = [line.strip() for line in header_lines if line.strip()]
+      self._header_lines = filter(None, [line.strip().replace(
+          'Number=G', 'Number=.').encode('utf-8') for line in header_lines])
+    else:
+      self._init_with_header(header_lines)
 
   def next(self):
-    text_line = next(self._text_lines).strip()
-    while not text_line:  # skip empty lines.
-      # This natively raises StopIteration if end of file is reached.
+    try:
       text_line = next(self._text_lines).strip()
+      while not text_line:  # skip empty lines.
```

Comment:
Can this not be a little function called next_non_empty_line()?  And then we can save a few lines of duplication and drop the comment?